src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -701,6 +719,26 @@ private ResultState AssertionStatusToResultState(AssertionStatus status)
         }
 
         /// <summary>
+        /// Adds a attachments element to a node and returns it.
+        /// </summary>
+        /// <param name="targetNode">The target node.</param>
+        /// <returns>The new attachments element.</returns>
+        private TNode AddAttatchmentsElement(TNode targetNode)
+        {
+            TNode attachmentsNode = targetNode.AddElement("attachments");
+
+            foreach (var attachment in TestAttachments)
+            {
+                var attachmentNode = attachmentsNode.AddElement("attachment", attachment.Filepath);
+
+                if (attachment.Description != null)
+                    attachmentNode.AddAttribute("description", attachment.Description);
```

Comment:
Description is a user supplied string that may be long or require CDATA encoding. Based on that, we should probably make it a child element rather than an attribute and CDATA encode it.

```C#
var attachmentNode = attachmentsNode.AddElement("attachment");
attachmentNode.AddElement("filename", attachment.Filepath);

if (attachment.Description != null)
    attachmentNode.AddElementWithCDATA("description", attachment.Description);
```

Tests will need to be changed accordingly.